letter-shop.md

```diff
@@ -0,0 +1,255 @@
+  plugin.on('outgoing_fulfill', function (transfer, fulfillment) {
+    console.log('outgoing fulfill', transfer, fulfillment, 'http://localhost:8000/' + fulfillment)
+    fetch('http://localhost:8000/' + fulfillment).then(function (inRes) {
+      return inRes.text()
+    }).then(function (body) {
+      pendingRes[transfer.id].end(body)
+    })
+  })
+
+  http.createServer(function (req, outRes) {
+    fetch('http://localhost:8000' + req.url).then(function (inRes) {
+      return inRes.text()
+    }).then(function (body) {
+      const parts = body.split(' ')
+      // Please send an Interledger payment to test.crypto.xrp.rrhnXcox5bEmZfJCHzPxajUtwdt772zrCW with amount 10 drops and condition Z8_mOwUpkAvI-DTGMKQ_kfCIh7QgcYGu0uuFgloGVr4
+      // 0      1    2  3           4       5  6                                                  7    8      9  10    11  12        13
+      if (parts[0] === 'Please') {
+        sendTransfer({
+          to: parts[6],
+          amount: parts[9],
+          executionCondition: parts[13]
+        }).then(function (transferId) {
+          console.log('transfer sent', transferId)
+          pendingRes[transferId] = outRes
+        }, function (err) {
+          console.error(err.message)
+        })
+      } else {
+        outRes.end(parts.join(' '))
+      }
+    })
+  }).listen(8001)
+})
+```
+
+Now run:
+```sh
+npm install node-fetch
+node ./shop.js # unless your shop was still running from before
+node ./proxy.js
+```
+and instead of visiting http://localhost:8000/, visit http://localhost:8001/ - you'll see that viewing the letter shop through the paying proxy is slower, but more convenient! You'll get a new letter each time you refresh the page.
+
+## Concepts learned
+
+The plugin used in all three scripts exposes the Ledger Plugin Interface (LPI) as described in [IL-RPC-4, draft 6](https://interledger.org/rfcs/0004-ledger-plugin-interface/draft-6.html), and of that, this script uses the following methods and events:
+* `sendTransfer` method (in `pay.js` and `proxy.js`, prepares a transfer to some other account on the same ledger)
+* `getInfo` method (used in `pay.js` and `proxy.js` to fill in the `ledger` field to pass to `sendTransfer`)
+* `getAccount` method (used in `pay.js` and `proxy.js` to fill in the `from` field to pass to `sendTransfer`)
+* `rejectIncomingTransfer` method (in `shop.js`, rejects an incoming transfer if someone tries to pay the wrong amount)
+* `fulfillCondition` method (in `shop.js`, fulfills the condition of an incoming transfer)
+* `incoming_prepare` event (in `shop.js`, is triggered when someone else sends you a conditional transfer)
+* `outgoing_fulfill` event (in `pay.js` and `proxy.js`, is triggered when someone else fulfills your conditional transfer)
+
+If you read the paragraphs above, you will have seen the following new words; see the glossary in [IL-RFC-19, draft 1](https://interledger.org/rfcs/0019-glossary/draft-1.html) as a reference if you
```

Comment:
I'm not sure it's necessary to include these terms at the bottom but if you want to have them it might be worth linking each to its definition in the glossary